src/Sticky.jsx

```diff
@@ -170,17 +170,23 @@ class Sticky extends Component {
     }
 
     handleResize (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
+
         winHeight = ae.resize.height;
         this.updateInitialDimension();
         this.update();
     }
 
     handleScrollStart (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
+        
         scrollTop = ae.scroll.top;
         this.updateInitialDimension();
     }
 
     handleScroll (e, ae) {
+        if (this.props.shouldFreeze()) { return; }
```

Comment:
Thinking I may want to optimize this a bit more, we shouldn't call this function with every scroll event... maybe I'll set state on scroll start?